signalfx/client.py

```diff
@@ -61,23 +62,48 @@ def send_event(self, event_type, category=None, dimensions=None,
         logging.debug('Sending event to SignalFx: %s', data)
         return data
 
+    @staticmethod
+    def update_metric(metric_type, description=None,
+                      custom_properties=None, tags=None):
+        data = {'type': metric_type.upper(),
+                'description': description or '',
+                'customProperties': custom_properties or {},
+                'tags': tags or []}
+        logging.debug('Sending metric metadata to SignalFx: %s', data)
+        return data
+
+    @staticmethod
+    def update_tag(description=None, custom_properties=None):
+        data = {'description': description or '',
+                'customProperties': custom_properties or {}}
+        logging.debug('Sending tag data to SignalFx: %s', data)
+        return data
+
 
 class SignalFxClient(BaseSignalFx):
     """SignalFx API client.
 
     This class presents a programmatic interface to SignalFx's metadata and
-    ingest APIs. At the time being, only ingest is supported; more will come
-    later.
+    ingest APIs. At the time being, ingest and retrieval of data
+    and functionality related to metadata and tags are supported.
+    More will come later.
     """
     _HEADER_API_TOKEN_KEY = 'X-SF-Token'
     _HEADER_USER_AGENT_KEY = 'User-Agent'
     _INGEST_ENDPOINT_DATAPOINT_SUFFIX = 'v2/datapoint'
     _INGEST_ENDPOINT_EVENT_SUFFIX = 'v2/event'
     _THREAD_NAME = 'SignalFxDatapointSendThread'
+    _METRIC_ENDPOINT_SUFFIX = 'v2/metric'
+    _DIMENSION_ENDPOINT_SUFFIX = 'v2/dimension'
+    _MTS_ENDPOINT_SUFFIX = 'v2/metrictimeseries'
+    _TAG_ENDPOINT_SUFFIX = 'v2/tag'
+    _ORGANIZATION_ENDPOINT_SUFFIX = 'v2/organization'
+    _TIME_SERIES_WINDOW_ENDPOINT_SUFFIX = 'v1/timeserieswindow'
```

Comment:
are we sure we want to expose v1 apis?  @tedoc2000  i thought this wasn't something we wanted to do?